hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/store/models/UpdatedHederaEvmAccount.java

```diff
@@ -15,31 +15,48 @@
  */
 package com.hedera.node.app.service.evm.store.models;
 
+import static org.apache.tuweni.units.bigints.UInt256.ZERO;
+
+import com.hedera.node.app.service.evm.store.contracts.HederaEvmEntityAccess;
+import edu.umd.cs.findbugs.annotations.Nullable;
 import java.util.Collections;
+import java.util.Map;
 import java.util.NavigableMap;
+import java.util.TreeMap;
 import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.bytes.Bytes32;
 import org.apache.tuweni.units.bigints.UInt256;
 import org.hyperledger.besu.datatypes.Address;
 import org.hyperledger.besu.datatypes.Hash;
 import org.hyperledger.besu.datatypes.Wei;
-import org.hyperledger.besu.evm.account.Account;
+import org.hyperledger.besu.evm.ModificationNotAllowedException;
 import org.hyperledger.besu.evm.account.AccountStorageEntry;
+import org.hyperledger.besu.evm.account.EvmAccount;
+import org.hyperledger.besu.evm.account.MutableAccount;
 
-public class UpdatedHederaEvmAccount implements Account {
-
+public class UpdatedHederaEvmAccount implements MutableAccount, EvmAccount {
+    protected Hash addressHash;
     private Address address;
     private long nonce;
     private Wei balance;
+    private HederaEvmEntityAccess hederaEvmEntityAccess;
+    protected final NavigableMap<UInt256, UInt256> updatedStorage;
+
+    @Nullable protected Bytes updatedCode;
+    @Nullable private Hash updatedCodeHash;
 
     public UpdatedHederaEvmAccount(Address address) {
-        this.address = address;
+        this(address, 0L, Wei.ZERO);
+        this.updatedCode = Bytes.EMPTY;
     }
 
     public UpdatedHederaEvmAccount(Address address, long nonce, Wei balance) {
         this.address = address;
+        this.addressHash = Hash.hash(address);
         this.nonce = nonce;
         this.balance = balance;
+        this.updatedStorage = new TreeMap<>();
+        this.updatedCode = null;
```

Comment:
Probably we can move line 50 here and always assign `updatedCode` as `Bytes.EMPTY`?